### ISSUE TITLE
STK fails to build with gcc 6

### ISSUE THREAD

**Issue by deveee** [supertuxkart/stk-code#2428]
As reported on Debian's bug tracker:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=811724

```
> sbuild (Debian sbuild) 0.67.0 (26 Dec 2015) on dl580gen9-02.hlinux
...
> cd /<<PKGBUILDDIR>>/obj-x86_64-linux-gnu/lib/bullet && /usr/bin/c++    -I/<<PKGBUILDDIR>>/lib/bullet/src  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -O2 -DNDEBUG   -o CMakeFiles/bulletdynamics.dir/src/BulletDynamics/ConstraintSolver/btHinge2Constraint.cpp.o -c /<<PKGBUILDDIR>>/lib/bullet/src/BulletDynamics/ConstraintSolver/btHinge2Constraint.cpp
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-128' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
>  4,0,38,1,4,0,39,1,4,0,40,1,67,0,41,1,};
>                                       ^
> 
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-124' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-56' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-8' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-32' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-104' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-44' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-52' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-12' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-60' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-64' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-92' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-128' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-127' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> make[4]: Entering directory '/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu'
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-126' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-125' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-124' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-123' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-122' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-121' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-120' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-119' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-118' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-117' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-116' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-115' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-114' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-128' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-127' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-126' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-125' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-124' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-123' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-122' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-121' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-120' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-119' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-118' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-117' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-116' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-115' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-114' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-113' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-112' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
> /<<PKGBUILDDIR>>/lib/bullet/src/LinearMath/btSerializer.cpp:415:38: error: narrowing conversion of '-111' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
...
```

**Comment by deveee** [comment]
We need probably something like this:
https://github.com/bulletphysics/bullet3/commit/1b7642f0f5954f5998bc946578a774369099232d

But it seems to be related also to:
https://github.com/bulletphysics/bullet3/commit/28de5ea03084c41e9418913100f513b108d036c8
and
https://github.com/bulletphysics/bullet3/commit/d99d9779a2d2733835d2a5087edff40b5de688c5

**Comment by deveee** [comment]
Fixed in 58846002ebf1137df372435140c1d91a5d455b25
